bundles/org.eclipse.orion.client.ui/web/orion/parameterCollectors.js

```diff
@@ -328,6 +318,24 @@ define(['i18n!orion/nls/messages', 'orion/webui/littlelib'],
 					localClose();
 					if (typeof(cancelFunction) === 'function') cancelFunction();
 				}, false);
+
+				if (commandInvocation.parameters.hasOptionalParameters()) {
```

Comment:
it was strange calling finish(self) and then executing more code to create buttons for the already-finished dialog. The alternative was returning early, but there was nothing sensible to return. Instead I moved the code to the end so that finish(self) is called after everything is set up, and we fall through to the single return point.